### PR TITLE
Implement tests for dimension roll up

### PR DESCRIPTION
**Pull request by kohrapha**
**Description:**
This CR implements tests and benchmarks for the `dimensionRollup` function.

### CODE REVIEW

File: exporter/awsemfexporter/metric_translator_test.go [hunk 1]
```diff
@@ -273,6 +273,98 @@ func TestCalculateRate(t *testing.T) {
...
+			"cloudwatch-otel",
+			nil,
+		},
+		{
+			"zero + single dim w/o instrumentation library name",
+			ZeroAndSingleDimensionRollup,
+			[]string{"a", "b", "c"},
+			noInstrumentationLibraryName,
+			[][]string{
+				{},
```
Comment (kohrapha) [inline exporter/awsemfexporter/metric_translator_test.go:340]: @mxiamxia or @shaochengwang: is this the desired output? Do we include `{}` as part of the output if we perform zero dim rollup with no instrumentation library name?

Comment (shaochengwang) [reply exporter/awsemfexporter/metric_translator_test.go:340]: yes. {} will be appended to dimensionArray. It indicates `Zero dimension`

Comment (kohrapha) [reply exporter/awsemfexporter/metric_translator_test.go:340]: I see, so what's the effect of `{}` vs just removing it entirely in the exported metrics?

Comment (mxiamxia) [reply exporter/awsemfexporter/metric_translator_test.go:1086]: if you remove `{}`, we can't generate a metric without dimension attaching which is what we want here

Comment (kohrapha) [reply exporter/awsemfexporter/metric_translator_test.go:1086]: Got it, thanks! Does this CR look good to both of you then?

Comment (shaochengwang) [reply exporter/awsemfexporter/metric_translator_test.go:1086]: Can you add a case for ZeroDimensionRollup for metric with no label (zero dimension)?

Comment (kohrapha) [reply exporter/awsemfexporter/metric_translator_test.go:1086]: @shaochengwang done!

**Review by shaochengwang — APPROVED** [review]
LGTM!

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**